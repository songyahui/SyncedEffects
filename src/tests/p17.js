"use hopscript"

var hh = require("hiphop");

var prg = <hh.module O>
  <hh.loop>
    <hh.local S1>
      <hh.sequence>
	<hh.if S1>
	  <hh.emit O/>
	</hh.if>
	<hh.pause/>
	<hh.emit S1/>
      </hh.sequence>
    </hh.local>
  </hh.loop>
</hh.module>;

exports.prg = new hh.ReactiveMachine(prg, "P17")
