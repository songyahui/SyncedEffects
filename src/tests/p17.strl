module P17:
output O;
loop
    signal S1 in
        present S1 then
            emit O
        end present;
        pause;
        emit S1
    end signal
end loop
end module
