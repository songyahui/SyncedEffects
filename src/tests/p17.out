P17> ;
--- Output:
P17> ;
--- Output:
P17> ;
--- Output:
